pkg/executables/cmk.go

```diff
@@ -120,6 +120,38 @@ func (c *Cmk) ValidateServiceOfferingPresent(ctx context.Context, zoneId string,
 	return nil
 }
 
+func (c *Cmk) ValidateDiskOfferingPresent(ctx context.Context, zoneId string, diskOffering v1alpha1.CloudStackResourceDiskOffering) error {
+	command := newCmkCommand("list diskofferings")
+	if len(diskOffering.Id) > 0 {
+		applyCmkArgs(&command, withCloudStackId(diskOffering.Id))
+	} else {
+		applyCmkArgs(&command, withCloudStackName(diskOffering.Name))
+	}
+	applyCmkArgs(&command, withCloudStackZoneId(zoneId))
+	result, err := c.exec(ctx, command...)
+	if err != nil {
+		return fmt.Errorf("getting disk offerings info - %s: %v", result.String(), err)
+	}
+	if result.Len() == 0 {
+		return fmt.Errorf("disk offering %s not found", diskOffering)
+	}
+
+	response := struct {
+		CmkDiskOfferings []cmkDiskOffering `json:"diskoffering"`
+	}{}
+	if err = json.Unmarshal(result.Bytes(), &response); err != nil {
+		return fmt.Errorf("failed to parse response into json: %v", err)
```

Comment:
```suggestion
		return fmt.Errorf("parsing response into json: %v", err)
```

Reply:
all other places while checking json response in file cmk.go use 'failed to parse ...'. 

Any particular reason to use 'parsing ...' ?

Reply:
rather redundant for a type error to say "failed to ....", we recently updated all our errors to remove it. So yea pls do update them all.

Reply:
update all `failed to parse' in cmk.go to `parsing`